core/client/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -140,11 +146,14 @@ protected FileInStream(URIStatus status, InStreamOptions options, FileSystemCont
     mShouldCache = mAlluxioStorageType.isStore();
     mShouldCachePartiallyReadBlock = options.isCachePartiallyReadBlock();
     mClosed = false;
-    mLocationPolicy = options.getLocationPolicy();
+    mCacheLocationPolicy = options.getLocationPolicy();
```

Comment:
I have made the change.  I think it will break REST API backward compatibility. Add @jsimsa  to take a look too.